website/blog/2023-10-26-TeachableAgent/index.mdx

```diff
@@ -0,0 +1,390 @@
+---
+title: AutoGen's TeachableAgent
+authors: rickyloynd-microsoft
+tags: [LLM, teach]
+---
+
+![Teachable Agent Architecture](img/teachable-arch.png)
+
+**TL;DR:**
+* We introduce **TeachableAgent** (which uses by **TextAnalyzerAgent**) so that users can teach their LLM-based assistants new facts, preferences, and skills.
+* We showcase examples of `TeachableAgent` learning and then recalling facts, preferences, and skills in later chats.
+
+
+## Introduction
+Conversational assistants based on LLMs can remember the current chat with the user, and can also demonstrate in-context learning of user teachings during the conversation. But the assistant's memories and learnings are lost once the chat is over, or when a single chat grows too long for the LLM to handle effectively. Then in subsequent chats the user is forced to repeat any necessary instructions over and over.
+
+`TeachableAgent` addresses these limitations by persisting user teachings across chat boundaries in long-term memory implemented as a vector database. Memory is automatically saved to disk at the end of each chat, then loaded from disk at the start of the next. Instead of copying all of memory into the context window, which would eat up valuable space, individual memories (called memos) are retrieved into context as needed. This allows the user to teach frequently used facts and skills to the teachable agent just once, and have it recall them in later chats.
+
+In order to make effective decisions about memo storage and retrieval, `TeachableAgent` calls an instance of `TextAnalyzerAgent` to analyze pieces of text in various ways. This adds extra LLM calls involving a relatively small number of tokens, which can add a few seconds to the time a user waits for each response.
```

Comment:
of `TextAnalyzerAgent` to analyze -->  of `TextAnalyzerAgent`, which is another AutoGen agent, to analyze

Reply:
fixed